packages/gamut/src/Banner/index.tsx

```diff
@@ -1,72 +1,91 @@
-import { CloseIcon } from '@codecademy/gamut-icons';
-import cx from 'classnames';
+import { MiniDeleteIcon } from '@codecademy/gamut-icons';
+import { variant } from '@codecademy/gamut-styles';
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
 import React from 'react';
 
-import { ButtonDeprecated, ButtonDeprecatedProps } from '../ButtonDeprecated';
-import styles from './styles.module.scss';
+import { IconButton, TextButton } from '../Button';
 
-export enum BannerStyle {
-  BorderBottom = 'border-bottom',
-  FullWidth = 'full-width',
-}
+type BannerVariants = 'navy' | 'yellow';
 
-export type BannerProps = {
-  classNames?: {
-    /**
-     * Class name for the container element.
-     */
-    container?: string;
-    /**
-     * Class name for the content wrapper
-     */
-    content?: string;
-  };
+export interface BaseBannerProps {
+  className?: string;
   /** Visual variations for banners */
-  displayStyle?: BannerStyle;
-  /**
-   * Whether or not the banner should be visible.
-   */
-  isClosed?: boolean;
-  /**
-   * Callback called when the user closes the banner.
-   */
-  onClose: ButtonDeprecatedProps['onClick'];
-  /**
-   * An icon or jsx element to be displayed to the left of the content.
-   */
-  icon?: React.ReactNode;
-};
+  variant?: BannerVariants;
+  /**  Callback called when the user closes the banner. */
+  onClose: () => void;
+}
 
-const BANNER_CLASSES = {
-  [BannerStyle.BorderBottom]: styles.containerBordered,
-  [BannerStyle.FullWidth]: styles.containerFullWidth,
-};
+export interface CTABanner extends BaseBannerProps {
+  /** Call to action text */
+  cta: string;
+  /** Link associated with CTA */
+  href: string;
+  /** Call to action click callback */
+  onCtaClick?: () => void;
+}
+
+export interface TextBanner extends BaseBannerProps {
+  cta?: never;
+  onCtaClick?: never;
+  href?: never;
+}
+
+type BannerProps = TextBanner | CTABanner;
+
+const BannerContainer = styled.div(
+  variant({
+    navy: { textColor: 'white', backgroundColor: 'navy' },
+    yellow: { textColor: 'navy', backgroundColor: 'yellow' },
+  }),
+  ({ theme }) => css`
+    display: grid;
+    width: 100%;
+    padding: ${theme.spacing[4]};
+    column-gap: ${theme.spacing[8]};
+    grid-template-columns: 2rem 1fr 2rem;
+    align-items: center;
+    text-align: center;
+
+    &:before {
+      content: '';
+    }
+  `
+);
 
 export const Banner: React.FC<BannerProps> = ({
   children,
-  classNames = {},
-  displayStyle = BannerStyle.FullWidth,
-  isClosed = false,
+  variant = 'navy',
+  cta,
+  href,
+  onCtaClick,
   onClose,
-  icon,
+  ...rest
 }) => {
-  if (isClosed) {
-    return null;
-  }
+  const mode = variant === 'navy' ? 'dark' : 'light';
+  const [button, iconButton] = [mode, `${mode}-alt`] as const;
```

Comment:
If `button === mode` there's no reason for `button`, right?

```suggestion
  const iconButton = `${mode}-alt`;
```